listenbrainz_spark/hdfs/upload.py

```diff
@@ -46,43 +47,48 @@ def process_json_listens(self, filename, data_dir, tmp_hdfs_path, schema):
             month = filename.split('/')[-1][0:-5]
             dest_path = os.path.join(data_dir, year, '{}.parquet'.format(str(month)))
 
+        if utils.path_exists(dest_path):
+            df.write.mode('append').parquet(dest_path)
+        else:
+            utils.save_parquet(df, dest_path)
+
         current_app.logger.info("Uploading to {}...".format(dest_path))
-        utils.save_parquet(df, dest_path)
         current_app.logger.info("File processed in {:.2f} seconds!".format(time.monotonic() - start_time))
 
-    def upload_mapping(self, archive, force=False):
+    def upload_mapping(self, archive):
         """ Decompress archive and upload mapping to HDFS.
 
             Args:
                 archive: Mapping tar file to upload.
                 force: If True deletes dir at path where mappings are to be uploaded.
         """
-        tmp_dump_dir = tempfile.mkdtemp()
         with tarfile.open(name=archive, mode='r:bz2') as tar:
-            self.upload_archive(tmp_dump_dir, tar, path.MBID_MSID_MAPPING, schema.msid_mbid_mapping_schema,
-                self.process_json, force=force)
+            with tempfile.TemporaryDirectory() as tmp_dump_dir:
+                self.upload_archive(tmp_dump_dir, tar, path.MBID_MSID_MAPPING, schema.msid_mbid_mapping_schema,
+                                    self.process_json, force=True)
 
     def upload_listens(self, archive, force=False):
         """ Decompress archive and upload listens to HDFS.
 
             Args:
                 archive: listens tar file to upload.
-                force: If True deletes dir at path where listens are to be uploaded.
+                force: If True deletes dir at path where listens are to be uploaded,
```

Comment:
We should consider renaming force here to `incremental` or something like that

Reply:
Renamed it to `overwrite`